ble-shelly-blu.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * This script will use BLE observer to listen for advertising data from nearby Shelly BLU devices, 
+ * decodes the data using a BTHome data structure, and emits the decoded data for further processing.
+ * 
+ * This script DOESN'T execute actions, only emit event. Can be used with ble-events-handler.js example.
+ * 
+ * All paramerts that the BLU devices can return:
+ * Each device will provide data solely from its sensors.
+ * - pid - packet ID
+ * - battery - the battery level of the device in %
+ * - temperature - the temperature value in °C if the device has temperature sensor 
+ * - humidity - the himidity value in % if the device has humidity sensor
+ * - illuminance - the illuminance value in lux if the device has light sensor
+ * - motion - 0/1 (motion/clear) if the device has motion sensor
+ * - window - 0/1 (close/open) if the device has reed switch
+ * - button - the number of presses if the device has button
+ * - rotation - the angle of rotatation in ° if the device has gyroscope
+ * - rssi - the signal strength is dB
+ * - address - The mac address of the Shelly BLU device
+ */
+
+/******************* START CHANGE HERE *******************/
+let CONFIG = {
+    // Specify the destination event where the decoded BLE data will be emitted. It allows for easy identification by other applications/scripts
+    eventName: "shelly-blu",
+
+    // When set to true, debug messages will be logged to the console
+    debug: false,
+};
+/******************* STOP CHANGE HERE *******************/
+
+let ALLTERCO_MFD_ID_STR = "0ba9";
+let BTHOME_SVC_ID_STR = "fcd2";
+
+let uint8 = 0;
+let int8 = 1;
+let uint16 = 2;
+let int16 = 3;
+let uint24 = 4;
+let int24 = 5;
+
+//Logs the provided message with an optional prefix to the console.
+function logger(message, prefix) {
+
+    //exit if the debug isn't enabled
+    if(!CONFIG.debug) {
+        return;
+    }
+
+    let finalText = "";
+
+    //if the message is list loop over it
+    if(Array.isArray(message)) {
+        for(let i = 0; i < message.length; i++) {
+            finalText = finalText + " " + JSON.stringify(message[i]);
+        }
+    }
+    else {
+        finalText = JSON.stringify(message);
+    }
+
+    //the prefix must be string
+    if(typeof prefix !== "string") {
+        prefix = "";
+    }
+    else {
+        prefix = prefix + ":"
+    }
+
+    //log the result
+    console.log(prefix, finalText);
+}
+
+// The BTH object defines the structure of the BTHome data
+let BTH = {};
+BTH[0x00] = { n: "pid", t: uint8 };
```

Comment:
can we add to this list all sensors and binary sensors that we expect - with motion and h&t are coming

Reply:
All are already added: 
```javascript
let BTH = {};
BTH[0x00] = { n: "pid", t: uint8 };
BTH[0x01] = { n: "battery", t: uint8, u: "%" };
BTH[0x02] = { n: "temperature", t: int16, f: 0.01, u: "tC" };
BTH[0x03] = { n: "humidity", t: uint16, f: 0.01, u: "%" };
BTH[0x05] = { n: "illuminance", t: uint24, f: 0.01 };
BTH[0x21] = { n: "motion", t: uint8 };
BTH[0x2d] = { n: "window", t: uint8 };
BTH[0x3a] = { n: "button", t: uint8 };
BTH[0x3f] = { n: "rotation", t: int16, f: 0.1 };
```